server/events/events.factory.ts

```diff
@@ -0,0 +1,53 @@
+import { Events } from "../../database/models/init-models"
+import { Sports } from "../../database/models/sports"
+import ValidationException from "../core/exceptions/validation.exception"
+import { EventRatingModel } from "./event-rating.model"
+import { EventModel } from "./event.model"
+
+export class EventFactory {
+
+	public static Create(name: string, date: Date, userId: number, sportId: number) {
+		return new EventModel(
+			undefined,
+			name,
+			date,
+			userId,
+			userId,
+			sportId,
+			"", // TODO: not correct, fix
```

Comment:
I wanted to avoid nullable fields in domain models - we should be pretty strict about preventing domain models from ever getting in an invalid state, with missing fields, especially if it's something crucial for business logic, or in this case, something that's shown in the UI. 

I've fixed it now by loading a `sport` with all the fields in service layer and then passing it to this method.